modules/account-lib/src/coin/dot/keyPair.ts

```diff
@@ -34,9 +34,19 @@ export class KeyPair extends Ed25519KeyPair {
     return createPair({ toSS58: keyring.encodeAddress, type: TYPE }, { secretKey, publicKey });
   }
 
-  /** @inheritdoc */
+  /**
+   // https://wiki.polkadot.network/docs/learn-accounts#address-format
+   * Returns the address in either mainnet polkadot format (starts with 1)
+   * or substrate format used for westend (starts with 5)
+   */
   getAddress(format?: AddressFormat): string {
-    return this.createPolkadotPair().address;
+    const substrateAddress = this.createPolkadotPair().address;
+    // generate polkadot (mainnet) address format
+    if (format && format === AddressFormat.polkadot) {
```

Comment:
Is there a way to use a similar pattern to the one in WP where we have a `tdot` controller and a `dot` controller? If so, the `tdot KeyPair` can override the `dot KeyPair` class method `getAddress`

Reply:
I see where you're going with this. Need to let this one simmer and see if there is a way to do it this way instead.